src/query/service/src/sessions/session_privilege_mgr.rs

```diff
@@ -231,6 +213,28 @@ impl SessionPrivilegeManager for SessionPrivilegeManagerImpl {
         )))
     }
 
+    #[async_backtrace::framed]
+    async fn validate_ownership(&self, object: &GrantObjectByID) -> Result<()> {
+        let role_mgr = RoleCacheManager::instance();
+        let tenant = self.session_ctx.get_current_tenant();
+
+        // if the object is not owned by any role, then considered as PUBLIC, which is always true
+        let owner_role = match role_mgr.find_object_owner(&tenant, object).await? {
+            Some(owner_role) => owner_role,
+            None => return Ok(()),
+        };
+
+        let available_roles = self.get_all_available_roles().await?;
+        if !available_roles.iter().any(|r| r.name == owner_role.name) {
+            return Err(ErrorCode::PermissionDenied(
+                "Permission denied, current session do not have the ownership of this object"
+                    .to_string(),
+            ));
+        }
+
+        Ok(())
+    }
+
```

Comment:
```suggestion
async fn validate_ownership(&self, object: &GrantObjectByID) -> Result<()> {
    let role_mgr = RoleCacheManager::instance();
    let tenant = self.session_ctx.get_current_tenant();

    if let Some(owner_role) = role_mgr.find_object_owner(&tenant, object).await? {
        let available_roles = self.get_all_available_roles().await?;
        if !available_roles.iter().any(|r| r.name == owner_role.name) {
            return Err(ErrorCode::PermissionDenied(
                "Permission denied, current session does not have the ownership of this object".to_string(),
            ));
        }
    }

    Ok(())
}
```

Reply:
I sometimes prefer to return early on some specified logic path in an function, thus the later code can do not care about the previous logics, sometimes this may make things easier to handle. quq

in this case, `if the object is not owned by any role, then considered as PUBLIC, which is always true` is the case which is ok to return eary by an `return Ok(())`, which means `everything is allowed on a object owned by PUBLIC`, but less obvious in the suggested version without comment.